packages/commons/src/interop-token/models.ts

```diff
@@ -3,134 +3,255 @@ import {
+  exp: z.number(),
+  nbf: z.number(),
+  iat: z.number(),
+  jti: z.string(),
+});
+export type InteropJwtCommonPayload = z.infer<typeof InteropJwtCommonPayload>;
 
 /* ========================================== 
     Interop CONSUMER Token 
   ========================================== */
-export type InteropJwtConsumerPayload = InteropJwtCommonPayload & {
-  client_id: ClientId;
-  sub: ClientId;
-  [PURPOSE_ID_CLAIM]: PurposeId;
-  digest?: ClientAssertionDigest;
-};
+export const InteropJwtConsumerPayload = InteropJwtCommonPayload.merge(
+  z.object({
+    client_id: ClientId,
+    sub: ClientId,
+    purposeId: PurposeId,
+    digest: ClientAssertionDigest.optional(),
+  })
+);
+export type InteropJwtConsumerPayload = z.infer<
+  typeof InteropJwtConsumerPayload
+>;
 
-export type InteropConsumerToken = {
-  header: InteropJwtHeader;
-  payload: InteropJwtConsumerPayload;
-  serialized: string;
-};
+export const InteropConsumerToken = z.object({
+  header: InteropJwtHeader,
+  payload: InteropJwtConsumerPayload,
+  serialized: z.string(),
+});
+export type InteropConsumerToken = z.infer<typeof InteropConsumerToken>;
 
-/* ========================================== 
-    Interop API Token 
-  ========================================== */
-export type InteropJwtApiCommonPayload = InteropJwtCommonPayload & {
-  client_id: ClientId;
-  sub: ClientId;
-  [ORGANIZATION_ID_CLAIM]: TenantId;
-};
+// ==========================================
+//     Interop M2M Token
+// ==========================================
+export const InteropJwtApiCommonPayload = InteropJwtCommonPayload.merge(
+  z.object({
+    client_id: ClientId,
+    sub: ClientId,
+    organizationId: TenantId,
+  })
+);
+export type InteropJwtApiCommonPayload = z.infer<
+  typeof InteropJwtApiCommonPayload
+>;
 
-export type InteropJwtApiM2MPayload = InteropJwtApiCommonPayload & {
-  [ROLE_CLAIM]: Extract<SystemRole, "m2m">;
-};
+export const InteropJwtApiM2MPayload = InteropJwtApiCommonPayload.merge(
+  z.object({ role: z.literal(systemRole.M2M_ROLE) })
+);
+export type InteropJwtApiM2MPayload = z.infer<typeof InteropJwtApiM2MPayload>;
 
-export type InteropJwtApiM2MAdminPayload = InteropJwtApiCommonPayload & {
-  [ROLE_CLAIM]: Extract<SystemRole, "m2m-admin">;
-  adminId: UserId;
-  // ^ ID of the admin user associated with the client
-};
+// ==========================================
+//   Interop M2M Admin Token
+// ==========================================
+export const InteropJwtApiM2MAdminPayload = InteropJwtApiCommonPayload.merge(
+  z.object({
+    role: z.literal(systemRole.M2M_ADMIN_ROLE),
+    adminId: z.string().uuid(),
+    // ^ ID of the admin user associated with the client
+  })
+);
+export type InteropJwtApiM2MAdminPayload = z.infer<
+  typeof InteropJwtApiM2MAdminPayload
+>;
 
+// ==========================================
+//   Interop API Token
+// ==========================================
 export type InteropJwtApiPayload =
```

Comment:
```suggestion
export type InteropJwtApiM2MPayload =
```

Reply:
Resolving, see https://github.com/pagopa/interop-be-monorepo/pull/1863#discussion_r2091315098